src/aura/BGE_DataImportBatchEntry/BGE_DataImportBatchEntryHelper.js

```diff
@@ -325,6 +325,30 @@
         component.set('v.data', rows);
     },
 
+    /**
+     * @description: Dry run all data import records for batch
+     */
+    massDryRun: function (component) {
+        this.showToast(component, $A.get('$Label.c.PageMessagesProcessing'), $A.get('$Label.c.bgeMassDryRunProcessing'), 'sticky');
+
+        let action = component.get('c.runMassDryRun');
+        action.setParams({
+            batchId: component.get('v.recordId')
+        });
+        action.setCallback(this, function (response) {
```

Comment:
Would this be worth using promises and defining the callback behavior within the promise? @kenslewis

Reply:
Yep, I think it would definitely be a good idea to return a Promise here.

Reply:
@ptewson-sfdo @kenslewis - What do ya'll think about something like this? I know we have pending changes on checkFieldsPermisssion, but I think I like the idea of having a 'utility' that both checkFieldsPermission and runMassDryRun (or really any) can use.

https://github.com/SalesforceFoundation/Cumulus/commit/c0be9a3f9c0b1d2f25608169fe601051b6dcdd16

Reply:
@psadie that approach is super cool - I really love that. The only issue is that there are about four other apex calls in that helper that now AREN'T going through the callApex function - it doesn't seem fair to sign you up for refactoring the entire form component, but are we ok bringing that new method in incrementally?

Reply:
Having chewed on this for a little bit, I think delivering incrementally makes sense rather than trying to do limitless refactor right now. But thoughts and discussion welcome!